### PR TITLE
 Fix the `this` argument used when checking a constructor

### PR DESCRIPTION
**Pull request by thomasspriggs**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- N/A The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- N/A My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_entry_point.cpp [hunk 1]
```diff
@@ -297,8 +297,8 @@ std::pair<code_blockt, std::vector<exprt>> java_build_arguments(
   const select_pointer_typet &pointer_type_selector,
   message_handlert &message_handler)
 {
-  const java_method_typet::parameterst &parameters =
-    to_java_method_type(function.type).parameters();
+  const java_method_typet function_type = to_java_method_type(function.type);
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_entry_point.cpp]: const ref?

**Review by smowton — APPROVED** [review]
Seems sensible

**Review by thk123 — APPROVED** [review]
Thanks!

**Comment by thomasspriggs** [comment]
Rebased on develop to see if it addresses the CI issues.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2e47537c3ba56d2dab425fae72e5324ddf8a9296).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/122875057

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f4f01cd58e8af7642a60f3d8d576de1bbc55e364).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/122950475

File: jbmc/src/java_bytecode/java_entry_point.cpp (hunk 1 above)
Comment (thomasspriggs) [reply jbmc/src/java_bytecode/java_entry_point.cpp]: Done.

**Pull request merged**